src/main/java/com/es/phoneshop/dao/impl/AbstractDao.java

```diff
@@ -0,0 +1,55 @@
+package com.es.phoneshop.dao.impl;
+
+import com.es.phoneshop.dao.Dao;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public abstract class AbstractDao<T> implements Dao<T> {
+
+    List<T> items = new ArrayList<>();
+
+    @Override
+    public synchronized void save(T item) {
+        if (item == null) {
+            throw new IllegalArgumentException("Item should not be null");
+        }
+        try {
```

Comment:
Throwing an exception during a happy path is a bad practice. Refactor this method to get rid of this